docs/DeveloperGuide.md

```diff
@@ -6,32 +6,215 @@
+
+**Step2.** If the number of provided arguments is less than the required number, the `toDo()` function will throw an `IllegalCommand` exception and return.  
+
+**Step3.** If no exception is thrown, two strings will be generated by parsing the user input and the `toDo()` function will pass the two arguments to `projectList` for creating and adding a `Todo` object to a `Project` object.  
+
+**Step4.** When `projectList` receives the call, it will get the target `Project` object and pass the description string to it.  
+
+**Step5.** The `Project` object will create a `Todo` object and add it to the `ArrayList<Todo>` in `Project` class.  
+
+#### View a Project
+
+The view function makes use of ```ProjectList``` and ```CommandHandler``` classes. It is facilitated by methods within the ```Todo``` and ```Project``` classes, stored within ```ProjectList```.
+Given below is an example usage scenario and how View Project behaves at each step
+
+**Step 1.** The user executes ```view cs2113``` to view the project titled cs2113. The command is retrieved and parsed by ```handleUserInput()``` in ```CommandHandler```, which calls ```view()``` in ```CommandHandler```
+
+**Step 2.** The ```view()``` method ensures that ```ProjectList``` has been correctly initialised and the command is valid, before calling ```view()``` in ```ProjectList```
+
+**Step 3.** ```view()``` in ```ProjectList``` calls ```findProjectIndex()``` in ```ProjectList```
+
+**Step 4.** ```findProjectIndex()``` loops through all ```Project``` objects in projectList until a ```Project``` with the same title as the one entered by the user is found. The ```Project```’s index is returned to ```CommandHandler```'s implementation of ```view()```
+
+**Step 5.** ```view()``` calls the ```printDetails()``` method of the ```Project``` at the given index
+
+**Step 6.** ```printDetails()``` displays project title and deadline to the user. It iterates through the ```Todo```s in the ```todos``` attribute and prints each `Todo` by calling `Todo`’s `toString()` function
+
+![image info](./UmlDiagrams/ViewProject.jpg)
+
+#### Add a Deadline to a Project
+
+**Step 1.** When `CommandHandler` receives a user input starting with string “adddeadline”, it will call the `addDeadline()` method
+
+**Step 2.** It will first check if the number of arguments provided matches what is required. If it is less, it will throw an `IllegalCommandException` and return.
+
+**Step 3.** If the number of arguments is valid, call the projectList’s `addDeadline()` method, with the arguments of title and deadline.
+
+**Step 4.** We call the `findProjectIndex()` method to find the given project that matches the title provided. If the value indicates that the project does not exist, return immediately and indicate no deadline was added.
+
+**Step 5.** If the index of the project was found, call that specific Project’s `setDeadline()` method and update the deadline.
+
+![image info](./UmlDiagrams/AddDeadline.jpg)
```

Comment:
It would be great to show to show a user icon to explicitly show that it is the user calling the start of the sequence diagrams, though I believe this might not be absolutely necessary.